File: tut/6mon/phones.pl
#!/usr/bin/perl -w

%names = ();
%streets = ();
%towns = ();
%phones = ();

open(PEOPLE, 'people.txt') or die('no file found');
while ($line = <PEOPLE>) {
    chomp $line;
    ($key, $name, $street, $town) = split(/,/, $line);
    $names{$key} = $name;
    $streets{$key} = $street;
    $towns{$key} = $town;
}

open(PHONES, 'phones.txt') or die ('no phones found');
while ($line = <PHONES>) {
    chomp $line;
    ($key, $type, $number) = split(/,/, $line);
    $phones{$key}{$type} = $number;
}

close PHONES;
close PEOPLE;

for $key (keys %names) {
    print "$names{$key}\n$streets{$key}, $towns{$key}\n";
    print "Phones: ";
    @to_print = ();
    if (defined($phones{$key})) {
        for $type (keys $phones{$key}) {
            push @to_print, "$phones{$key}{$type} ($type)";
        }
        print join(", ", @to_print);
    }
    print "\n\n";
}

File: tut/6mon/phones.py
#!/usr/bin/python2.7

names = {}
streets = {}
phones = {}
towns = {}

with open("people.txt", "r") as f:
  for line in f:
    (key, name, street, town) = line.strip().split(",")
    names[key] = name
    streets[key] = street
    towns[key] = town
    phones[key] = {}

with open("phones.txt", "r") as f:
  for line in f:
    (key, type, number) = line.strip().split(",")
    phones[key][type] = number

for key in names:
  print names[key]
  print "%s, %s" % (streets[key], towns[key])
  phone_strings = []
  for type in phones[key]:
    phone_strings.append(phones[key][type])
  print ", ".join(phone_strings)
  print ""
